File: lecture-practice-code/java-to-kotlin-library-app/src/main/java/com/group/javatokotlinlibraryapp/dto/user/response/UserResponse.java
package com.group.javatokotlinlibraryapp.dto.user.response;


import com.group.javatokotlinlibraryapp.domain.user.User;

public class UserResponse {

    private final long id;
    private final String name;
    private final Integer age;

    public UserResponse(User user) {
        this.id = user.getId();
        this.name = user.getName();
        this.age = user.getAge();
    }

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public Integer getAge() {
        return age;
    }

}

File: lecture-practice-code/java-to-kotlin-library-app/src/main/kotlin/com/group/javatokotlinlibraryapp/dto/user/response/UserResponse.kt
package com.group.javatokotlinlibraryapp.dto.user.response

import com.group.javatokotlinlibraryapp.domain.user.User

data class UserResponse(
    val id: Long,
    val name: String,
    val age: Int?
) {

    companion object {
        fun of(user: User): UserResponse {
            return UserResponse(
                id = user.id!!,
                name = user.name,
                age = user.age
            )
        }
    }
}